go/kv/consul.go

```diff
@@ -35,6 +36,12 @@ func NewConsulStore() KVStore {
 	if config.Config.ConsulAddress != "" {
 		consulConfig := consulapi.DefaultConfig()
 		consulConfig.Address = config.Config.ConsulAddress
+		
+		if strings.HasPrefix(config.Config.ConsulAddress,"https://") {
+		  consulConfig.Scheme = "https"
+		  consulConfig.Address = strings.Replace(config.Config.ConsulAddress,"https://","",9)
```

Comment:
what does `9` stand for? Did you mean for :unlimited"? Use `-1` as per https://golang.org/pkg/strings/#Replace

Reply:
the 9 are the amount of characters of `https://` .

Reply:
`n` is the number of replacements, not the number of characters. Please change to `-1`.